### ISSUE TITLE
Feature request: is it possible to implemente a `Classification` in `LinXGBoost`?

### ISSUE THREAD

**Issue by bjmwang** [ldv1/LinXGBoost#1]
And will it perform better?

**Comment by ldv1** [comment]
If I remember well, for binary classification, it is just a matter of changing the objective function. I will have a look at it if I have some time. Will it perform better? It certainly depends on the problems at hand.

**Comment by bjmwang** [comment]
1. In [sklearn's GBDT](https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/ensemble/gradient_boosting.py#L9), regression and classification (binary or multi) only differ in the loss function.

2. Yes, as it always is.
In fact, I meant the average  performance over a set of various problems. Sorry for the ambiguities.

**Comment by ldv1** [comment]
Implementation for binary classification is done. I just added a test. I do not know at the moment how to extend it to multiclass. At the leaves, we have scores, and if the score is positive, then the label is 1 whereas if the score is negative, the label is 0. The extension to multiclass is thus not straightforward. I will look at sklearn implementation.
